### PR TITLE
create event registration form

### LINKED ISSUE #93: Create event registration page

### PR DESCRIPTION
**Pull request by Sarah-Addo**
Screenshot: https://screenshot.googleplex.com/1m5ufnNCr80

### CODE REVIEW

File: src/main/webapp/open_hours.js [hunk 1]
```diff
@@ -45,36 +50,51 @@ class TimeOption {
...
-      this.dayOpenInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("show-time-area");
-    } else if (!this.organizationDay.propertyMap.isOpen) {
-      this.dayClosedInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("hide-time-area");
-    } else {
-      this.dayOpenInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("show-time-area");
+
+    // Only show open closed options for organization modification
```
Comment (jnnybk) [inline src/main/webapp/open_hours.js]: I'm not sure if having this for specific pages and not for other pages is the best way to solve reusing the TimeOption class, if we decide to go with this option.

For example, I was thinking about having the Event registration page having:

(MM/DD/YYYY)
Date                    Time(s)
07/20/2020         3:15 - 4:15
                           6:15 - 7:15
07/21/2020         5:00 - 7:00

Comment (Sarah-Addo) [reply src/main/webapp/open_hours.js]: Right now it im designing it mvp style which would mean that the event would only have one day as an option so it is easier to implement. But they can have multiple times for that day

**Review by andrewm898 — APPROVED** [review]

File: src/main/webapp/open_hours.js [hunk 2]
```diff
@@ -45,36 +50,51 @@ class TimeOption {
...
+    if (!document.getElementById("event-registration-form-area")) {
```
Comment (belraquib) [inline src/main/webapp/open_hours.js]: Generally special casing if a certain id is available is a bad code smell.

You have several options here; if you want to do so: 
   move this code into a :   renderIntoElement(element) {}  function, that will allow you decide to create the elements in the calling code. You can also extract a new class that does this instead that encapsulates the data as well as creates the elements.

**Review by belraquib — CHANGES_REQUESTED** [review]

File: src/main/webapp/register_event.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+				<input class = "registration-entry" type = "text" name = "event-state"/>
+				<label>Event Zip Code: </label>
+				<input class = "registration-entry" type = "text" name = "event-zip-code"/>
+				<label>Contact Email: </label>
+				<input class = "registration-entry" type = "text" name = "event-email"/>
+				<label>Contact Phone Number: </label>
+				<input class = "registration-entry" type = "text" pattern = "[0-9]{10}" name = "event-phone-num"/>
+				<label>Event Details: </label>
+				<textarea class = "registration-entry" name ="event-details" maxlength = "180"></textarea>
+				<label>Event Partners: </label>
```
Comment (jnnybk) [inline src/main/webapp/register_event.html:39]: I think we should also have an Event Owners field, based on our Data Structuring powerpoint discussion!

File: src/main/webapp/register_event.html [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+				<input class = "registration-entry" type = "text" name = "event-street-address"/>
+				<label>Event City:</label>
+				<input class = "registration-entry" type = "text" name = "event-city"/>
+				<label>Event State:</label>
+				<input class = "registration-entry" type = "text" name = "event-state"/>
+				<label>Event Zip Code: </label>
+				<input class = "registration-entry" type = "text" name = "event-zip-code"/>
+				<label>Contact Email: </label>
+				<input class = "registration-entry" type = "text" name = "event-email"/>
+				<label>Contact Phone Number: </label>
```
Comment (jnnybk) [inline src/main/webapp/register_event.html:37]: We should have a Contact Name

File: src/main/webapp/open_hours.js (hunk 2 above)
Comment (belraquib) [reply src/main/webapp/open_hours.js]: Let's continue to think about this:

I believe we talked about the options of : extract creating these open/close options in a new function (and call it elsewhere!) or pass in a constructor argument that tells this object whether or not to create these elements.

File: src/main/webapp/open_hours.js [hunk 5]
```diff
@@ -45,36 +46,51 @@ class TimeOption {
...
-      this.dayOpenInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("show-time-area");
-    } else if (!this.organizationDay.propertyMap.isOpen) {
-      this.dayClosedInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("hide-time-area");
-    } else {
-      this.dayOpenInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("show-time-area");
+
+    // Only show open closed options for organization modification
```
Comment (belraquib) [inline src/main/webapp/open_hours.js]: update the comment here

File: src/main/webapp/open_hours.js [hunk 6]
```diff
@@ -17,24 +17,25 @@ document.addEventListener("DOMContentLoaded", async function() {
...
+    const saturdayTimeOption = new TimeOption("Saturday", /*forRegistration*/ true, null, optionArea, /*showOpenClosedOptions*/ true);
+    const sundayTimeOption = new TimeOption("Sunday", /*forRegistration*/ true, null, optionArea, /*showOpenClosedOptions*/ true);
   }
 });
 
 class TimeOption {
   // organization.hoursOpen[index(1-7 for day of week)].propertyMap.
   // fromToPairs.value[index(how many set of hours for that day )].propertyMap.from/to
-  constructor(day, forRegistration, organizationDay, optionArea) {
+  constructor(day, forRegistration, organizationDay, optionArea, showOpenClosedOptions) {
```
Comment (belraquib) [inline src/main/webapp/open_hours.js:33]: Leave a small comment for each parameter.

**Review by belraquib — APPROVED** [review]

**Pull request merged**